### PR TITLE
Bug fix: Wrong ouptut size due to wrong amount of padding in transposed convolutions with odd stride

### PR DESCRIPTION
**Pull request by ecobost**
Made 4 changes (in order of importance):
* 0c48ac85640f7a23171aaf2c38048e9885bade06 fixes the issue of the model not producing the same output length as the input. Curently, for instance, the example in the readme will receive a 1-sec input (48000 samples) and return a 47936 output (for discrete model). Fix is tested in pytorch 2.1+cu12.1 for both discrete and continuous models. It will work in most cases (particularly for any examples where num_samples in input is divisible by hop_length), especifically, if the conv stride is even and input_length (to the conv) is divisible by the stride, or stride is odd and input_length + 1 is not divisible by it.
* 825d106319960841bc7addbca71f8d5a281d4b71 disables the VAE sampling if the model is in eval mode. Seems logical to have repeatable reconstructions.
* f9392111116b124f943a125c66677362b9475c30 makes it mypy compliant
* 7382cb2f37980b22ce3302ffd5af7425fb0bd53f gets rid of global imports (replaces Tensor with torch.Tensor, PretrainedModel -> transformers.Pretrainedmodel and PretrainedConfig-> transformers.PretrainedConfig)

### CODE REVIEW

**Comment by elyxlz** [comment]
hi, thanks for your contribution. I'm going to close this as the first change has been implemented with another pr, the second one is unorthodox (as all diffusion model training runs use sampled latents as ground truth), maybe I would consider it as a flag. The second and third are not important for a repo of this size to me. Feel free to fork!
